openresty/openresty.go

```diff
@@ -4,9 +4,13 @@ import (
 	"math"
 	"strconv"
 	"strings"
+	"regexp"
 )
 
 func Name(version string) string {
+	pattern := regexp.MustCompile(`([0-9]+)(\.[0-9]+)?(\.[0-9]+)?(\.[0-9]+)?`)
```

Comment:
**regexp.MustCompile()** should be called at startup by **init()** in this file. Because **regexp.MustCompile()** panics if the expression cannnot be parsed.

refs: https://golang.org/pkg/regexp/#MustCompile

Reply:
Thanks for the review @cubicdaiya. Should we call `MustCompile` in `init` or use `Compile` instead, also returning the default name (`ngx_openresty`) in case of error? IMO there's not much advantage in using `MustCompile` in this case. Thoughts?

Reply:
Isn't calling `regexp.Compile()` in `init()` simpler? If `regexp.Compile()` is called, we need to handling error like below. It looks a little verbose.

```go
name, err := openresty.Name(builder.Version)
if err != nil {
    // error handling
    // name is "ngx_openresty"
}
```

Reply:
Agreed and pushed. It looks more idiomatic to use init.